mmdeploy/codebase/mmpose/deploy/pose_detection_model.py

```diff
@@ -98,10 +98,12 @@ def forward(self,
         inputs = inputs.contiguous().to(self.device)
         batch_outputs = self.wrapper({self.input_name: inputs})
         batch_outputs = self.wrapper.output_to_list(batch_outputs)
-        if self.model_cfg.model.type == 'YOLODetector':
-            return self.pack_yolox_pose_result(batch_outputs, data_samples)
 
         codec = self.model_cfg.codec
+
+        if codec.type == 'YOLOXPoseAnnotationProcessor':
```

Comment:
@Ben-Louis hi, this two lines should be put after line 108 as the codec maybe list of dict as in [mspn](https://github.com/open-mmlab/mmpose/blob/8102383761e877317825ef0d5e234ca4d584e5bc/configs/body_2d_keypoint/topdown_heatmap/coco/td-hm_mspn50_8xb32-210e_coco-256x192.py#L35)
```
Traceback (most recent call last):
  File "/root/workspace/mmdeploy/mmdeploy/utils/utils.py", line 41, in target_wrapper
    result = target(*args, **kwargs)
  File "/root/workspace/mmdeploy/mmdeploy/apis/visualize.py", line 72, in visualize_model
    result = model.test_step(model_inputs)[0]
  File "/usr/local/lib/python3.8/dist-packages/mmengine/model/base_model/base_model.py", line 145, in test_step
    return self._run_forward(data, mode='predict')  # type: ignore
  File "/usr/local/lib/python3.8/dist-packages/mmengine/model/base_model/base_model.py", line 340, in _run_forward
    results = self(**data, mode=mode)
  File "/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/root/workspace/mmdeploy/mmdeploy/codebase/mmpose/deploy/pose_detection_model.py", line 104, in forward
    if codec.type == 'YOLOXPoseAnnotationProcessor':
AttributeError: 'list' object has no attribute 'type'
```

Reply:
Thanks. This bug has been fixed in the latest commit